app/controllers/katello/api/v2/activation_keys_controller.rb

```diff
@@ -305,13 +306,36 @@ def find_content_view_environments
         find_cve_for_single
       elsif params[:content_view_environments] || params[:content_view_environment_ids]
         @content_view_environments = ::Katello::ContentViewEnvironment.fetch_content_view_environments(
-              labels: params[:content_view_environments],
-              ids: params[:content_view_environment_ids],
-              organization: @organization || @activation_key&.organization)
+          candlepin_names: params[:content_view_environments],
+          ids: params[:content_view_environment_ids],
+          organization: @organization || @activation_key&.organization)
+        if @content_view_environments.blank?
+          handle_errors(candlepin_names: params[:content_view_environments],
+          ids: params[:content_view_environment_ids])
+        end
       end
+      handle_blank_cve_params
       @organization ||= @content_view_environments.first&.organization
     end
 
+    def handle_blank_cve_params
+      # Activation keys do not require CVEs to be associated. So it's possible the user intends to clear them.
+      if params.key?(:environment_id) && params[:environment_id].blank? && params.key?(:content_view_id) && params[:content_view_id].blank?
+        @content_view_environments = []
+      elsif params.key?(:content_view_environments) && params[:content_view_environments].blank?
+        @content_view_environments = []
+      elsif params.key?(:content_view_environment_ids) && params[:content_view_environment_ids].blank?
+        @content_view_environments = []
+      end
+    end
+
+    def update_cves?
```

Comment:
Should this be moved to mutli cv params handling thingie, so that both hosts and aks can use it . Or are they wildly different between AK and Hosts ?